esrally/utils/repo.py

```diff
@@ -103,7 +103,7 @@ def update(self, distribution_version):
 
     def _find_matching_tag(self, distribution_version):
         tags = git.tags(self.repo_dir)
-        for version in versions.versions(distribution_version):
+        for version, _ in versions.VersionVariants(distribution_version).all_versions:
```

Comment:
Would it make sense to expose this as a function instead? This would maybe read a bit nicer:

```python
for version, _ in versions.variants_of(distribution_version):
    ...
```

Reply:
The class `VersionVariants` is used in `utils/versions.py` leveraging state. `repo.py` doesn't require state, just the list expansion of all versions.
Are you suggesting `variants_of()` could be a utility function within `util/versions.py` that does the object instantiation/calling of `.all_versions()` or?

Reply:
Yes, I am proposing a utility function along the lines of:

```python
def variants_of(distribution_version):
    for version, _ in versions.VersionVariants(distribution_version).all_versions
        yield version
```

This implementation would even get rid of the superfluous second return value of `#all_versions()` and give us the chance to keep the class `VersionVariants` internal to to the `versions` module.

Reply:
Ok fixed in 13ba1ad7dce96530611619a4eb8d3999a3640074